### PR TITLE
#165020085 rating articles

### PR DESCRIPTION
**Pull request by Karlmusingo**
#### What does this PR do?
Add the rating functionality for articles
#### Description of Task to be completed?
- Rating of articles,
- Displaying ratings for articles
#### How should this be manually tested?
- `/`
- `articles/:slug`
#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

N/A

#### Screenshots (if appropriate)
image

image

image
#### Questions:

N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,8 @@ yarn.lock
 
 /cypress/videos
 
+/cypress/videos
+
```
Comment (gracelungu) [inline .gitignore]: @Karlmusingo Why is this path duplicated?

File: cypress/integration/articles/create.spec.js [hunk 2]
```diff
@@ -105,4 +105,21 @@ describe('Create article', () => {
...
+      response: {
+        data: {
+          message: 'rating submitted successfully',
+        },
+      }, // and force the response to be: []
+    });
+    cy.get('input[name="rate"]')
+      .first()
+      .check({ force: true });
+  });
```
Comment (gracelungu) [inline cypress/integration/articles/create.spec.js:149]: @Karlmusingo Where is the assertion for the above test?

File: src/Components/Articles/TopArticle.jsx [hunk 3]
```diff
@@ -78,11 +75,12 @@ const TopArticle = props => {
               <p className="">{readTime}</p>
             </div>
             <div className="rating">
-              {[...Array(rate)].map((e, i) => (
+              {/* {[...Array(rate)].map((e, i) => (
                 <i key={i}>
                   <Icon icon={faStar} />
                 </i>
-              ))}
+              ))} */}
```
Comment (gracelungu) [inline src/Components/Articles/TopArticle.jsx]: @Karlmusingo Could you please remove the commented code above?

File: src/Components/Rating/Rating.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          3 stars
+        </label>
+        <input type="radio" id="star2" name="rate" value="2" onChange={this.ratingChange} />
+        <label htmlFor="star2" title="text">
+          2 stars
+        </label>
+        <input type="radio" id="star1" name="rate" value="1" onChange={this.ratingChange} />
+        <label htmlFor="star1" title="text">
+          1 star
+        </label>
```
Comment (gracelungu) [inline src/Components/Rating/Rating.jsx]: @Karlmusingo This could have been better to be dynamic.

File: src/__test__/actions/rating.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      }),
+    );
+
+    await store.dispatch(rating(4, 'article_slug'));
+    // const actions = store.getActions();
+
+    expect.assertions(0);
+    // expect(actions[0].type).toEqual('RATING_ERROR');
+    // expect(actions[0].payload).toEqual({ message: 'karl' });
+  });
```
Comment (gracelungu) [inline src/__test__/actions/rating.test.js:42]: @Karlmusingo Can you please remove the commented code above?

**Review by gracelungu — CHANGES_REQUESTED** [review]

**Review by dusmel — APPROVED** [review]

**Review by gracelungu — APPROVED** [review]

**Comment by kipropbrian** [comment]
The content of this commit `Feature(Social Login): add social login functionalities ` is totally different from the title. Whats up there?

**Pull request merged**